tests/unit/test_finder.py

```diff
@@ -34,6 +34,15 @@ def test_no_partial_name_match(data):
     assert found.url.endswith("gmpy-1.15.tar.gz"), found
 
 
+@patch('pip.index.os.path.exists')
+def test_tilde(data):
```

Comment:
And a comment as to why it's needed (because ```~```-expansion is only done for files that exist) would be good too.